vms/evm/sync/customrawdb/accessors_state_sync_test.go

```diff
@@ -0,0 +1,507 @@
+	it = NewSyncStorageTriesIterator(db, nil)
+	defer it.Release()
+	count := 0
+	for it.Next() {
+		count++
+	}
+	require.NoError(t, it.Error())
+	require.Equal(t, 0, count)
+}
+
+func TestClearAllSyncStorageTries(t *testing.T) {
+	db := rawdb.NewMemoryDatabase()
+	root := common.HexToHash("0xabc")
+	// Keys that should be cleared
+	require.NoError(t, WriteSyncStorageTrie(db, root, common.HexToHash("0x01")))
+	require.NoError(t, WriteSyncStorageTrie(db, root, common.HexToHash("0x02")))
+	// Key that should not be cleared due to wrong length.
+	bad := make([]byte, 0, len(syncStorageTriesPrefix)+2*common.HashLength+1)
+	bad = append(bad, syncStorageTriesPrefix...)
+	bad = append(bad, root.Bytes()...)
+	bad = append(bad, common.HexToHash("0xff").Bytes()...)
+	bad = append(bad, byte(0x00))
+	require.NoError(t, db.Put(bad, []byte("x")))
+
+	require.NoError(t, ClearAllSyncStorageTries(db))
+
+	// Only the malformed key should remain
+	count := 0
+	it := db.NewIterator(syncStorageTriesPrefix, nil)
+	defer it.Release()
+	for it.Next() {
+		count++
+	}
+	require.NoError(t, it.Error())
+	require.Equal(t, 1, count)
+}
+
+func TestClear_NoKeys(t *testing.T) {
+	root := common.HexToHash("0xabc")
+	cases := []struct {
+		name  string
+		clear func(db ethdb.KeyValueStore) error
+		iter  func(db ethdb.Iteratee) ethdb.Iterator
+	}{
+		{
+			name:  "segments_no_keys",
+			clear: func(db ethdb.KeyValueStore) error { return ClearSyncSegments(db, root) },
+			iter:  func(db ethdb.Iteratee) ethdb.Iterator { return NewSyncSegmentsIterator(db, root) },
+		},
+		{
+			name:  "storage_no_keys",
+			clear: func(db ethdb.KeyValueStore) error { return ClearSyncStorageTrie(db, root) },
+			iter:  func(db ethdb.Iteratee) ethdb.Iterator { return NewSyncStorageTriesIterator(db, nil) },
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			db := rawdb.NewMemoryDatabase()
+
+			require.NoError(t, tc.clear(db))
+
+			it := tc.iter(db)
+			defer it.Release()
+			count := 0
+			for it.Next() {
+				count++
+			}
+			require.NoError(t, it.Error())
+			require.Equal(t, 0, count)
+		})
+	}
+}
+
+func TestSyncPerformedAndLatest(t *testing.T) {
+	db := rawdb.NewMemoryDatabase()
+
+	require.NoError(t, WriteSyncPerformed(db, 10))
+	require.NoError(t, WriteSyncPerformed(db, 20))
+	require.NoError(t, WriteSyncPerformed(db, 15))
+
+	// Iterator yields all
+	it := NewSyncPerformedIterator(db)
+	defer it.Release()
+	var vals []uint64
+	for it.Next() {
+		vals = append(vals, UnpackSyncPerformedKey(it.Key()))
+	}
+	require.NoError(t, it.Error())
+	require.ElementsMatch(t, []uint64{10, 20, 15}, vals)
+
+	// Latest is max
+	latest, err := GetLatestSyncPerformed(db)
+	require.NoError(t, err)
+	require.Equal(t, uint64(20), latest)
+}
+
+func TestGetLatestSyncPerformedEmpty(t *testing.T) {
+	db := rawdb.NewMemoryDatabase()
+	latest, err := GetLatestSyncPerformed(db)
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), latest)
```

Comment:
I think `require.Zero` also works here as well (I thought our linter would catch this - so maybe I'm wrong?)